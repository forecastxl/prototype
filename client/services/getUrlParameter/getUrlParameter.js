const getUrlParameter = parameterName => {
  // remove questionmark from querystring
  const query = window.location.search.substring(1)

  // split all parameters
  const params = query.split('&')

  for (let i = 0; i < params.length; i += 1) {
    // split parameters into key value pairs
    const pair = params[i].split('=')

    // return value if it matches the desired parameterName
    if (pair[0] === parameterName) return pair[1]
  }

  return false
}

export default getUrlParameter
