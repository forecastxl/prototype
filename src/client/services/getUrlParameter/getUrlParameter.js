const getUrlParameter = parameterName => {
  // Remove questionmark from querystring
  const query = window.location.search.substring(1)

  // Split all parameters
  const params = query.split('&')

  for (let i = 0; i < params.length; i += 1) {
    // Split parameters into key value pairs
    const pair = params[i].split('=')

    // Return value if it matches the desired parameterName
    if (pair[0] === parameterName) return pair[1]
  }

  return false
}

export default getUrlParameter
