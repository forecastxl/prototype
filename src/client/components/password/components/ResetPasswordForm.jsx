import React, { PropTypes } from 'react'
import { Field, reduxForm } from 'redux-form'
import { onSubmitActions } from 'redux-form-submit-saga'
import RaisedButton from 'material-ui/RaisedButton'
import { Alert } from '../../alert'
import { TextField } from '../../form'
import getUrlParameter from '../../../services/getUrlParameter'

function DumbResetPasswordForm({ handleSubmit, pristine, submitting, error }) {
  const token = getUrlParameter('token')

  if (!token) {
    return <div>Could not find a token. Did you use the correct link?</div>
  }

  return (
    <form onSubmit={handleSubmit}>
      {error && <Alert message={error} type={'error'} />}
      <Field name="password" component={TextField} label="Password" type="password" fullWidth />
      <Field
        name="passwordConfirmation"
        component={TextField}
        label="Repeat password"
        type="password"
        fullWidth
      />
      <RaisedButton
        type="submit"
        label="Reset password"
        primary
        fullWidth
        disabled={pristine || submitting}
      />
    </form>
  )
}

DumbResetPasswordForm.propTypes = {
  error: PropTypes.string,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired
}

DumbResetPasswordForm.defaultProps = {
  error: ''
}

export default reduxForm({
  form: 'reset-password',
  onSubmit: onSubmitActions(
    'RESET_PASSWORD',
    'RESET_PASSWORD_SUCCESS',
    'RESET_PASSWORD_VALIDATION_FAILURE'
  )
})(DumbResetPasswordForm)
