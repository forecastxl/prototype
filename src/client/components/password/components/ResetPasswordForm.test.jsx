import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import getUrlParameter from '../../../services/getUrlParameter'
import { DumbResetPasswordForm } from './ResetPasswordForm'

jest.mock('../../../services/getUrlParameter', () => jest.fn())

describe('<DumbResetPasswordForm />', () => {
  it('warns of a missing token', () => {
    getUrlParameter.mockImplementationOnce(() => false)
    const wrapper = shallow(
      <DumbResetPasswordForm handleSubmit={() => {}} pristine submitting={false} />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly', () => {
    getUrlParameter.mockImplementationOnce(() => 'token')
    const wrapper = shallow(
      <DumbResetPasswordForm handleSubmit={() => {}} pristine submitting={false} />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
