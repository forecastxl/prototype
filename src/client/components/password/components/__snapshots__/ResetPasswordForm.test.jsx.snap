// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbResetPasswordForm /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    fullWidth={true}
    label="Password"
    name="password"
    type="password"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Repeat password"
    name="passwordConfirmation"
    type="password"
  />
  <RaisedButton
    disabled={true}
    fullWidth={true}
    label="Reset password"
    labelPosition="after"
    primary={true}
    secondary={false}
    type="submit"
  />
</form>
`;

exports[`<DumbResetPasswordForm /> warns of a missing token 1`] = `
<div>
  Could not find a token. Did you use the correct link?
</div>
`;
